CorsixTH/Lua/room.lua

```diff
@@ -427,27 +434,27 @@ function Room:onHumanoidEnter(humanoid)
   end
 end
 
+----- BEGIN Save game compatibility -----
+-- These function are merely for save game compatibility.
+-- For 0.69 gamesaves and below.
+-- And they does not participate in the current game logic.
+-- Do not move or edit
 function Room:createDealtWithPatientCallback(humanoid)
   self.dealt_patient_callback = --[[persistable:room_dealt_with_patient_callback]] function (staff_humanoid)
     if not staff_humanoid.waiting_on_other_staff then
       return
     end
-    local staff_member = self:getStaffMember()
-    if staff_member then
-      staff_member:setNextAction(self:createLeaveAction())
-      staff_member:queueAction(MeanderAction())
-      staff_member:setMood("staff_wait", "deactivate")
-      staff_member:setDynamicInfoText("")
-    end
     staff_humanoid:setCallCompleted()
     staff_humanoid.waiting_on_other_staff = nil
-    self:commandEnteringStaff(staff_humanoid)
-    self:setStaffMember(staff_humanoid)
+    staff_humanoid:setNextAction(self:createLeaveAction())
+    staff_humanoid:queueAction(MeanderAction())
+    staff_humanoid:setMood("staff_wait", "deactivate")
+    staff_humanoid:setDynamicInfoText("")
     self.waiting_staff_member = nil
     self.dealt_patient_callback = nil
   end
-  self.waiting_staff_member = humanoid
```

Comment:
This is the only point `waiting_staff_member` is set inside the room code. However, other parts of the file depend on checking this variable. One part of `gp.lua` also depends on this variable.